google/resource_compute_global_address_test.go

```diff
@@ -29,6 +31,29 @@ func TestAccComputeGlobalAddress_basic(t *testing.T) {
 	})
 }
 
+func TestAccComputeGlobalAddress_ipv6(t *testing.T) {
+	var addr computeBeta.Address
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeGlobalAddressDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeGlobalAddress_ipv6,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeBetaGlobalAddressExists(
```

Comment:
You could use https://golang.org/pkg/net/#ParseIP

ip := ParseIP(address)
ip.To16 != nil // means it is a IPv6 address.
ip.To4 != nil // means this is a IPv4 address.

Checking if the IpVersion match what we expect is probably fine for the IPv6 case. My main goal was not to test if the API "lies" to us but rather make sure we don't create an ipv6 address when we don't use the beta feature (and since we don't have the ip version field in that case, the only way to check is looking at address itself to validate it is ipv4).